kotlin-audio/src/main/java/com/doublesymmetry/kotlinaudio/notification/NotificationManager.kt

```diff
@@ -426,97 +500,56 @@ class NotificationManager internal constructor(
+                    is NotificationButton.STOP -> {
+                        showStopButton = true
                     }
-                }.build().apply {
-                    setColor(config.accentColor ?: Color.TRANSPARENT)
-                    config.smallIcon?.let { setSmallIcon(it) }
-                    for (button in buttons) {
-                        if (button == null) continue
-                        when (button) {
-                            is NotificationButton.PLAY_PAUSE -> {
-                                showPlayPauseButton = true
-                            }
-                            is NotificationButton.STOP -> {
-                                showStopButton = true
-                            }
-                            is NotificationButton.FORWARD -> {
-                                showForwardButton = true
-                                showForwardButtonCompact = button.isCompact
-                            }
-                            is NotificationButton.BACKWARD -> {
-                                showRewindButton = true
-                                showRewindButtonCompact = button.isCompact
-                            }
-                            is NotificationButton.NEXT -> {
-                                showNextButton = true
-                                showNextButtonCompact = button.isCompact
-                            }
-                            is NotificationButton.PREVIOUS -> {
-                                showPreviousButton = true
-                                showPreviousButtonCompact = button.isCompact
-                            }
-                            else -> {}
-                        }
+
+                    is NotificationButton.FORWARD -> {
+                        showForwardButton = true
+                        showForwardButtonCompact = button.isCompact
+                    }
+
+                    is NotificationButton.BACKWARD -> {
+                        showRewindButton = true
+                        showRewindButtonCompact = button.isCompact
+                    }
+
+                    is NotificationButton.NEXT -> {
+                        showNextButton = true
+                        showNextButtonCompact = button.isCompact
+                    }
+
+                    is NotificationButton.PREVIOUS -> {
+                        showPreviousButton = true
+                        showPreviousButtonCompact = button.isCompact
                     }
-                    setMediaSessionToken(mediaSession.sessionToken)
-                    setPlayer(player)
+
+                    else -> {}
                 }
+            }
+        }
     }
 
     fun hideNotification() = scope.launch {
-        internalNotificationManager?.setPlayer(null)
+        flow<Unit> {
+            internalNotificationManager?.setPlayer(null)
+            internalNotificationManager = null
+            invalidate()
+            delay(500L)
```

Comment:
It is not obvious to me why this delay is here.. In general I am a bit hesitant about introducing delays because the amount of waiting time often depends on the speed of your device.

Reply:
This delay was because of the comment of Peyman's statement in the issue and to make sure the notification is hidden. But If you merge your fix in this branch
https://github.com/doublesymmetry/KotlinAudio/pull/85/files/f761c8697967563dced131765d73150bdd7b4571#diff-7d05808a134fc6efdaef2d0049ac5bf23ee4367c6c2601f159b4577f6bbb29bfR448
I think we can remove this.